### PR TITLE
Entry/Profile Form Updates

### LINKED ISSUE #1059: Update style of pulse entry forms
Need to update the Pulse page forms for the 'profile' and 'entry' pages:

Desktop:
https://redpen.io/ede8cba6116ad234e2
https://redpen.io/xhd5029bc3a098fdcc

Mobile:
https://redpen.io/cmd2fe6c2b0d8b4b3e
https://redpen.io/uvd52362e1ecc9003a

cc @youriwims @mmmavis Let me know if you need any more guidance for these. All files are in Sketch.

@gideonthomas @xmatthewx

### PR DESCRIPTION
**Pull request by youriwims**
Closes #1059

https://network-pulse-staging-pr-1115.herokuapp.com/add
https://network-pulse-staging-pr-1115.herokuapp.com/myprofile

### CODE REVIEW

**Comment by youriwims** [comment]
@natalieworth I know you already reviewed this as another pr before, but if you could review it again to make sure all the updates are correct?

File: components/form-fields/dynamic-checkbox-group.jsx [hunk 1]
```diff
@@ -75,7 +75,6 @@ DynamicCheckboxGroup.propTypes = {
 };
 
 DynamicCheckboxGroup.defaultProps = {
-  colNum: 1,
```
Comment (alanmoo) [inline components/form-fields/dynamic-checkbox-group.jsx]: Why’d this get removed? The prop is still referenced in the file.

File: pages/add/form/detail-info-fields.js [hunk 2]
```diff
@@ -27,30 +27,38 @@ module.exports = {
...
   issues: {
     type: IssuesField,
     label: <IssuesLabel />,
     colCount: 1
+    
   },
   tags: {
     type: Tags,
     label: `Tags: Comma separated. Spaces are ok. Issues are added automatically.`,
+    labelClassname: `form-label`,
```
Comment (alanmoo) [inline pages/add/form/detail-info-fields.js]: Why is this so WET? If you’re adding this class to every label, is there another way to do it?

Comment (alanmoo) [reply pages/add/form/detail-info-fields.js]: (Not a blocker, just...curious)

File: pages/profile-edit/profile-edit.scss [hunk 3]
```diff
@@ -1,19 +1,65 @@
 .profile-edit-page {
+  h5.email {
+    &::after {
+      @include form-helper-text("(email won't be displayed or shared)");
```
Comment (alanmoo) [inline pages/profile-edit/profile-edit.scss]: This is fine for now but last I checked, pseudoelement content isn’t accessible.

File: pages/add/add.scss [hunk 4]
```diff
@@ -51,15 +59,47 @@
 
   fieldset.thumbnail {
     .help-text {
-      margin-left: 1rem;
-      color: $gray-light;
+      @include form-helper-text("");
```
Comment (alanmoo) [inline pages/add/add.scss]: Why include this if it’s empty?

File: pages/add/add.scss [hunk 5]
```diff
@@ -51,15 +59,47 @@
...
     }
 
     input[type="button"] {
       @include thumbnail-button-style();
     }
   }
 
+  fieldset.get_involved {
+    textarea {
+      height: 4.625rem;
```
Comment (alanmoo) [inline pages/add/add.scss]: Why REMs here? Seems like EMs would be better, as it’s a direct mapping to “lines of text to fit”.

File: pages/profile-edit/form/fields.js [hunk 6]
```diff
@@ -33,58 +30,53 @@ module.exports = function(showLongBio) {
...
+      labelClassname: `h5-heading bio`,
       fieldClassname: `form-control`,
       validator: [validator.maxLengthValidator(140)],
       charLimit: 140,
       charLimitText: function(charCount, charLimit) {
         return charLimit - charCount;
       }
     },
-    twitter: {
+    Twitter: {
```
Comment (mmmavis) [inline pages/profile-edit/form/fields.js]: This changes the field's `name` attribute from `twitter` to `Twitter`. Are we doing this due to API changes? If not, this should not be changed.

File: components/form-fields/dynamic-checkbox-group.jsx (hunk 1 above)
Comment (youriwims) [reply components/form-fields/dynamic-checkbox-group.jsx]: Thanks for pointing this out—I'm going to remove the rest of the prop references.

File: pages/profile-edit/form/fields.js [hunk 7]
```diff
@@ -33,58 +30,53 @@ module.exports = function(showLongBio) {
...
       type: `text`,
       label: `Twitter`,
       placeholder: `https://twitter.com/username`,
+      labelClassname: `h5-heading`,
       fieldClassname: `form-control`,
       validator: [validator.urlValidator(`Twitter`)]
     },
-    linkedin: {
+    Linkedin: {
```
Comment (mmmavis) [inline pages/profile-edit/form/fields.js]: Same question -  Are we doing this due to API changes? If not, this should not be changed.

File: pages/profile-edit/form/fields.js [hunk 8]
```diff
@@ -33,58 +30,53 @@ module.exports = function(showLongBio) {
...
       type: `text`,
-      label: `LinkedIn`,
+      label: `Linkedin`,
       placeholder: `https://linkedin.com/in/username`,
+      labelClassname: `h5-heading`,
       fieldClassname: `form-control`,
       validator: [validator.urlValidator(`LinkedIn`)]
     },
-    github: {
+    Github: {
```
Comment (mmmavis) [inline pages/profile-edit/form/fields.js]: Same question -  Are we doing this due to API changes? If not, this should not be changed.

File: pages/profile-edit/profile-edit.scss (hunk 3 above)
Comment (mmmavis) [reply pages/profile-edit/profile-edit.scss]: Let's stick with the existing way of adding help text 
e.g.,

```jsx
user_bio: {
      type: `textarea`,
      label: (
        <div>
          Bio <em>(do not include any personal information)</em>
        </div>
      ),
     ...
```

You can give the `em` element a class name if you think you need to target and style the help text.

Comment (youriwims) [reply pages/profile-edit/profile-edit.scss]: welp, that's not good. Let me double check that.

File: pages/profile-edit/form/fields.js (hunk 6 above)
Comment (youriwims) [reply pages/profile-edit/form/fields.js]: Okay, that's fine. But, how is the `.twitter` class added on this element?

File: pages/add/add.scss (hunk 5 above)
Comment (youriwims) [reply pages/add/add.scss]: `rem`s were originally used to establish `textarea` height in this file, so primarily for consistency. I can update them to `em`s though.

File: pages/add/form/detail-info-fields.js (hunk 2 above)
Comment (youriwims) [reply pages/add/form/detail-info-fields.js]: Thing is, some `label`s have another element inside of them. Let me check it out and see if there's some improvements I can make.

Comment (youriwims) [reply pages/add/form/detail-info-fields.js]: After checking, I think this is the best approach. The class is only added to specific labels that share these styles. I can't really target these labels without also targeting the checkbox labels.

**Comment by natalieworth** [comment]
@youriwims I think the only thing I'm noticing (and I'm not sure if it actually affects mobile) is on this page https://network-pulse-staging-pr-1115.herokuapp.com/add
I am able to scroll to the right also?

**Comment by natalieworth** [comment]
Screen Shot 2019-07-16 at 1 52 00 PM

**Comment by youriwims** [comment]
@natalieworth, no, scrolling isn't good 😬 But, I've tried all screen sizes(320px+) and can't get my screen to scroll.

**Comment by youriwims** [comment]
@alanmoo @mmmavis are you getting the overflow scroll?

**Comment by natalieworth** [comment]
@youriwims I don't see it anymore LOL HOW

**Comment by youriwims** [comment]
@natalieworth 👻

File: pages/profile-edit/profile-edit.jsx [hunk 9]
```diff
@@ -169,49 +169,44 @@ class ProfileEdit extends React.Component {
...
           <div className="mb-4">
-            <div className="mb-3">
-              <span>
-                Email <em>(email won't be displayed or shared)</em>
+            <h5 className="h5-heading email mb-1">
+              Email <span className="hint-text">(email won't be displayed or shared)</span>
+            </h5>
+            <div className="d-flex flex-column flex-md-row">
+              <p className="mb-1">{user.email}</p>
+              <span className="d-block d-sm-inline-block ml-0 ml-md-4">
```
Comment (mmmavis) [inline pages/profile-edit/profile-edit.jsx]: This is within a flexbox so `block` or `inline-block` doesn't make much difference. To make things slightly more semantic we can change this wrapper from

```html
<span className="d-block d-sm-inline-block ml-0 ml-md-4">...</span>
```

to

```html
<div className="ml-md-4">...</div>
```

File: pages/add/add.jsx [hunk 10]
```diff
@@ -195,52 +195,54 @@ class Add extends React.Component {
...
           Please keep your language simple and useful for a broad audience. No
           jargon. Submissions may be lightly edited by our curators for
           spelling, grammar and style consistency.
         </p>
-        <div className="mb-4">
-          <h2>Basic Info</h2>
-          <div className="posted-by">
-            <p className="d-inline-block mr-3 mb-3">
-              Posted by: <span className="text-muted">{user.name}</span>
+        <div className="pt-4 mb-3">
```
[the remark this answers is not in the document]
Comment (youriwims) [reply pages/add/add.jsx]: cool cool, good point.

File: pages/add/add.jsx [hunk 11]
```diff
@@ -195,52 +195,54 @@ class Add extends React.Component {
...
+          <h5 className="h5-heading pt-4 mb-2">Get Help</h5>
           <Form
             ref="getHelpForm"
             fields={getHelpFields}
             inlineErrors={true}
             onUpdate={updateCallback}
           />
-          <div className="submit-section">
-            <p>
+          <div className="submit-section mt-5">
```
[the remark this answers is not in the document]
Comment (youriwims) [reply pages/add/add.jsx]: yeah, definitely.

**Comment by youriwims** [comment]
@natalieworth, did you intend for the _Posted By_ **name** below to follow the form text rules ↙️? If not, we should probably find a type style from our style guide 😬 
- Mobile
  - Font Size: 16px
  - Line Height: 24px
- Desktop
  - Font Size: 20px, 
  - Line Height: 28px

Screen Shot 2019-07-30 at 10 50 11 AM

File: pages/add/add.scss [hunk 12]
```diff
@@ -1,6 +1,8 @@
 .add-page {
   .posted-by {
-    margin-top: $form-field-vertical-margin / 2;
+    p:first-child {
+      @include scaleText(16px,24px,20px,28px);
```
[the remark this answers is not in the document]
Comment (youriwims) [reply pages/add/add.scss]: Asked Nat for clarification on this 👉  https://github.com/mozilla/network-pulse/pull/1115#issuecomment-516457041

File: pages/add/add.scss [hunk 13]
```diff
@@ -51,15 +59,41 @@
...
     }
 
     input[type="button"] {
       @include thumbnail-button-style();
     }
   }
 
+  fieldset.get_involved {
+    textarea {
+      height: 4.625rem;
```
[the remark this answers is not in the document]
Comment (youriwims) [reply pages/add/add.scss]: It appears as though you're correct 😅 🙈

File: scss/overrides/_variables.scss [hunk 14]
```diff
@@ -8,9 +8,13 @@ $bp-xl: #{map-get($grid-breakpoints, xl)}; // >= 1200px
 
 // spacing
 
-$form-field-vertical-margin: 30px;
+$form-field-vertical-margin: 24px;
 $form-check-input-margin-x: 0.5rem;
 
+//sizing
```
[the remark this answers is not in the document]
Comment (youriwims) [reply scss/overrides/_variables.scss]: haha for sure.

**Comment by natalieworth** [comment]
> Screen Shot 2019-07-30 at 10 50 11 AM
> 
> @natalieworth, did you intend for the _Posted By_ **name** above to follow the form text rules ↙️? If not, we should probably find a type style from our style guide 😬
> 
>     * Mobile
>       
>       * Font Size: 16px
>       * Line Height: 24px
> 
>     * Desktop
>       
>       * Font Size: 20px
>       * Line Height: 28px

@youriwims We could use same text style as the "Posted by" so it doesn't have to be in the grey

**Comment by youriwims** [comment]
> @youriwims We could use same text style as the "Posted by" so it doesn't have to be in the grey

Sure, but I mainly meant the font-size + line height Lol 😬

**Comment by youriwims** [comment]
> > @youriwims We could use same text style as the "Posted by" so it doesn't have to be in the grey
> 

Sure, but I mainly meant the font-size + line height Lol 😬 @natalieworth

**Comment by natalieworth** [comment]
> > > @youriwims We could use same text style as the "Posted by" so it doesn't have to be in the grey
> 
> Sure, but I mainly meant the font-size + line height Lol 😬 @natalieworth

ooh whoops. I think I intended it to use the body large style. Could we do body-large so we stay consistent with the style guide (sry if I changed it in the Sketch doc) :D

**Pull request merged**